src/main/java/clare/Task/Deadline.java

```diff
@@ -0,0 +1,83 @@
+package clare.task;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Objects;
+
+import clare.exception.StringConvertExceptions;
+
+/**
+ * Represents a task type with a deadline
+ */
+public class Deadline extends Todo {
+    private final LocalDate deadlineDate;
+    private LocalTime deadlineTime;
+
+    /**
+     * Constructor for Deadline class
+     * @param title the task title
+     * @param deadline the deadline of the task
+     * @param isDone the status
+     * @throws StringConvertExceptions if the format is invalid
+     */
+    public Deadline(String title, String deadline, boolean isDone) throws StringConvertExceptions {
+        super(title, isDone);
+        try {
+            String[] d = deadline.split(" ");
+            if (Objects.equals(d[0], "now")) {
+                deadlineDate = LocalDate.now();
+                this.deadlineTime = LocalTime.now();
+                return;
+            }
+            deadlineDate = LocalDate.parse(d[0]);
+            if (d.length > 1) {
+                deadlineTime = LocalTime.parse(d[1]);
+            }
+        } catch (DateTimeParseException | StringIndexOutOfBoundsException e) {
+            throw new StringConvertExceptions("Error deadline format: "
+                    + deadline + " Please follow this format YYYY-MM-DD HH:MM");
+        }
+    }
+
+    @Override
+    String getTypeString() {
+        return "D";
+    }
+
+    @Override
+    public TaskType getType() {
+        return TaskType.D;
+    }
+
+    /**
+     * Checks if the date give is same as the task deadline
+     * @param date the given date to compare
+     * @return true if equals, false otherwise
+     */
```

Comment:
```suggestion
    /**
     * Checks if the date give is same as the task deadline
     * @param date the given date to compare
     * @return true if equals, false otherwise
     */
```
Would the description and parameters be more easily readable with a empty line between them? I've noticed this issue in several other parts as well.